backend/src/schema/resolvers/reports.spec.js

```diff
@@ -163,8 +187,7 @@ describe('report', () => {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-14T08:19:29Z" title="Monday, October 14th 2019, 10:19:29 am +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -287,3 +291,227 @@ describe('report', () => {
     })
   })
 })
+
+describe('reports query', () => {
+  let query
+  let mutate
+  let authenticatedUser = null
+  let moderator
+  let user
+  let author
+  const categoryIds = ['cat9']
+
+  const reportMutation = gql`
+    mutation($resourceId: ID!, $reasonCategory: String!, $reasonDescription: String!) {
+      report(
+        resourceId: $resourceId
+        reasonCategory: $reasonCategory
+        reasonDescription: $reasonDescription
+      ) {
+        type
+      }
+    }
+  `
+  const reportsQuery = gql`
+    query {
+      reports(orderBy: createdAt_desc) {
+        createdAt
+        reasonCategory
+        reasonDescription
+        submitter {
+          id
+        }
+        resourceId
+        type
+        user {
+          id
+        }
+        post {
+          id
+        }
+        comment {
+          id
+        }
+      }
+    }
+  `
+
+  beforeAll(() => {
+    const { server } = createServer({
+      context: () => {
+        return {
+          driver,
+          user: authenticatedUser,
+        }
+      },
+    })
+    query = createTestClient(server).query
+    mutate = createTestClient(server).mutate
+  })
+
+  beforeEach(async () => {
+    authenticatedUser = null
+
+    moderator = await factory.create('User', {
+      id: 'mod1',
+      role: 'moderator',
+      email: 'moderator@example.org',
+      password: '1234',
+    })
+    user = await factory.create('User', {
+      id: 'user1',
+      role: 'user',
+      email: 'test@example.org',
+      password: '1234',
+    })
+    author = await factory.create('User', {
+      id: 'auth1',
+      role: 'user',
+      name: 'abusive-user',
+      email: 'abusive-user@example.org',
+    })
+    await instance.create('Category', {
+      id: 'cat9',
+      name: 'Democracy & Politics',
+      icon: 'university',
+    })
+
+    await Promise.all([
+      factory.create('Post', {
+        author,
+        id: 'p1',
+        categoryIds,
+        content: 'Not for you',
+      }),
+      factory.create('Post', {
+        author: moderator,
+        id: 'p2',
+        categoryIds,
+        content: 'Already seen post mention',
+      }),
+      factory.create('Post', {
+        author: user,
+        id: 'p3',
+        categoryIds,
+        content: 'You have been mentioned in a post',
```


what do these posts have to do with mentions? copy and pasted maybe?

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-14T15:24:07Z" title="Monday, October 14th 2019, 5:24:07 pm +02:00">Oct 14, 2019</time>_

---


😂 Yes, unreflected copy and paste …
Thanks for you pointing that out !!! 😘
Changed locally yet …